courses/signals_test.py

```diff
@@ -0,0 +1,42 @@
+"""
+Tests for signals
+"""
+from unittest.mock import patch
+from django.db.models.signals import post_save
+from django.test import TestCase
+from factory.django import mute_signals
+
+from courses.factories import CourseRunFactory, CourseRunCertificateFactory, UserFactory
+
+
+# pylint: disable=unused-argument
+@patch("courses.signals.transaction.on_commit", side_effect=lambda callback: callback())
+class CourseCertificateTests(TestCase):
```

Comment:
any particular reason for not supporting class-based? 
actually, I was thinking in terms of encapsulation.

Reply:
Primary reason is just consistency. The rest of the codebase is in function-based pytest style, so it's very helpful to stay consistent with the way we write tests (we have linting rules and a formatter to maintain consistency as well).

As for why we prefer pytest-style over unittest-style, that's a deeper conversation. Two of the main reasons have to do with boilerplate code (function-based tests simply require less code) and data/object dependencies in test cases (sharing data between unittest-style test classes often becomes an inheritance nightmare that is difficult to make sense of, whereas the dependencies for pytest-style test cases are pretty obvious since they are right there in the function arguments).

Regarding encapsulation, the fact that these tests are in a separate module is all the encapsulation we need. Regarding `setup`, a pytest [fixture](https://docs.pytest.org/en/latest/fixture.html) can be used for that exact purpose. Here's just [one very simple example](https://github.com/mitodl/mitxpro/blob/master/ecommerce/mail_api_test.py#L33-L36) from our test suite. Any test case that needs a `Company` object can just add `company` to its function arguments and it will be created and made available in the function body. 

In general I can't recommend learning & adopting pytest enough. It has its flaws but it's a massive improvement over the unittest approach